### ISSUE TITLE
Create option to humanize time output 

### ISSUE THREAD

**Issue by mmarras** [hgrecco/pint#2080]
This touches on #1459. 

Again, I am (again) trying to get a human friendly version of a time. Somewhat with the same intention why https://pypi.org/project/humanize/ exists. 

Background is I want to show time data in a human friendly way. k, M etc are not helpful here.

102123123 s, very few humans will be able to comprehend. I can do ureg.Quantity('102123123 s').to('a') and get 3.23... a. Nice but I will have to know apriory what the meaningful time-unit is here.

I am proposing to create a modification `to_compact` which will automatically convert units of dimension [time] to a suitable humanfriendly time unit. Suitable time unit is the time unit which leaves the magnitude with at least one but the least amount of integer parts.

Example: 

1001 ms should convert to 1.001 s
59 s should still be 59 s
but 60 s should be 1.0 min 
61 min should be 1.01... h 
and so on. 
I'm not sure if pint implements a decade but it does implement a century  
so finally 
ureg.Quantity('10231231232s ').to_compact(humantime=True) -> 3.23 century
(again millenium seem not to be implemented, eon is, maybe that's another topic to pick up: https://en.wikipedia.org/wiki/Unit_of_time, but not to dilute this post). 

As demonstrated above, `to_compact()`, nor `to_preferred()` nor `to_base_units()` are helpful but don't get the job done out of the box.

**Comment by mmarras** [comment]
So I guess one way to go about is to get all units of dimension [time] in the registry. Then rank them by “size” and then start at the lowest one until we get the first non-zero integer part. 
would that be flexible enough?

**Comment by andrewgsavage** [comment]
Yes. I think that could pick up too many units, but it's a good starting
point. It would be good to be able to set the list of units to rank after
registry init too.
